spec/components/hiring_staff/vacancy_form_page_heading_component_spec.rb

```diff
@@ -0,0 +1,125 @@
+require "rails_helper"
+
+RSpec.describe HiringStaff::VacancyFormPageHeadingComponent, type: :component do
+  let(:organisation) { create(:school, name: "Teaching Vacancies Academy") }
+  let(:state) { "create" }
+  let(:vacancy) { create(:vacancy, state: state, job_title: "Test job title") }
+  let(:session_job_location) { nil }
+  let(:session_readable_job_location) { nil }
+  let(:session_vacancy_attributes) { { "job_location" => session_job_location, "readable_job_location" => session_readable_job_location } }
+  let(:current_step) { 99 }
+
+  subject { described_class.new(vacancy, session_vacancy_attributes) }
+
+  before do
+    allow(subject).to receive(:current_step).and_return(current_step)
+    allow(subject).to receive(:total_steps).and_return(100)
+    allow(subject).to receive(:current_organisation).and_return(organisation)
+    vacancy.organisation_vacancies.create(organisation: organisation) if vacancy.present?
+    render_inline(subject)
+  end
+
+  describe "#show_current_step?" do
+    context "when vacancy state is create or review" do
+      let(:state) { "create" }
+
+      it "shows the current step" do
+        expect(rendered_component).to include(I18n.t("jobs.current_step", step: current_step, total: 100))
+      end
+    end
+
+    context "when vacancy state is not create or review" do
+      let(:state) { "copy" }
+
+      it "does not show the current step" do
+        expect(rendered_component).not_to include(I18n.t("jobs.current_step", step: 100, total: 101))
+      end
+    end
+  end
+
+  describe "#heading" do
+    context "when there is no vacancy object" do
+      let(:vacancy) { nil }
+
+      context "when the user is not signed in as a single school and the current step is 1" do
+        let(:organisation) { create(:trust) }
+        let(:current_step) { 1 }
+        let(:session_job_location) { "at_one_school" }
+        let(:session_readable_job_location) { "That Specific School" }
+
+        it "returns the create a job title without an organisation, ignoring the session parameters" do
+          expect(
+            page.text.split("Step").first.strip,
```

Comment:
Why do we do this bit like this as opposed to `rendered_component`?

Reply:
I want to avoid a false positive where the incorrect heading `Create a job for X School` passes a test for `Create a job`.
The way I did it was split the rendered text where it goes onto the next content ('Step') to make sure the first line _only_ includes `Create a job`.